packages/core/src/common/refs.ts

```diff
@@ -78,25 +71,10 @@ export function setRef<T extends HTMLElement>(refTarget: IRef<T> | undefined, re
 export function refHandler<T extends HTMLElement, K extends string>(
     refTargetParent: { [k in K]: T | null },
     refTargetKey: K,
-): IRefCallback<T>;
-export function refHandler<T extends HTMLElement, K extends string>(
-    refTargetParent: { [k in K]: T | IRefObject<T> | null },
```

Comment:
This type made it pretty easy to cause runtime errors since it does not require the passed type to include `IRefObject<T>`. The problem here being that `k` here should be contravariant, but since typescript doesn't allow for this it defaults to being covariant.

This kind of typing issue is pretty common with refs and Typescript, and leads to things like the bivariance hack in the base react types.

More detail here: https://github.com/palantir/blueprint/issues/4611